doc/application/application.rst

```diff
@@ -223,105 +621,142 @@ Below is an example CMakeList.txt:
+   :file:`zephyr/.config` in the build directory). This can be used
+   during development as described below in :ref:`override_kernel_conf`.
+
+#. The application's configuration file(s) given by the
+   :makevar:`CONF_FILE` variable, either as set explicitly by the user
+   or using one of the default values detailed below.
 
-#. The application's current configuration. (i.e. The :file:`.config` file.)
+#. The board's default configuration for the current :makevar:`BOARD`
+   setting (i.e. the :file:`boards/ARCHITECTURE/BOARD/BOARD_defconfig`
+   file in the Zephyr base directory).
 
-#. The application's default configuration. (Specified by ``CONF_FILE``, i.e. :file:`prj.conf`.)
+#. The kernel's default configuration settings (i.e. the default value given to
+   the option in one of Zephyr's :file:`Kconfig` files).
 
-#. The board configuration used by the application.
-   (i.e. The board's :file:`_defconfig` file.)
+The Zephyr build system determines a value for :makevar:`CONF_FILE` by
+checking the following, in order:
 
-#. The kernel's default configuration.
-   (i.e. One of the kernel's :file:`Kconfig` files.)
+- Any value given to :makevar:`CONF_FILE` in your application
+  :file:`CMakeLists.txt`, passed to the the CMake command line, or present
+  in the CMake variable cache, takes precedence.
+
+- If a CMake command, macro, or function ``set_conf_file`` is defined, it
+  will be invoked and must set :makevar:`CONF_FILE`.
+
+- If the file :file:`prj_BOARD.conf` exists in your application directory,
+  where ``BOARD`` is the BOARD value set earlier, it will be used.
+
+- Finally, if :file:`prj.conf` exists in your application directory, it will
+  be used.
+
+If :makevar:`CONF_FILE` specifies multiple files, they will be merged in order.
 
 For information on available kernel configuration options, including
 inter-dependencies between options, see the :ref:`configuration`.
 
-Default Board Configuration
-===========================
+.. _application_set_conf:
 
-An application's :file:`.conf` file defines its default kernel configuration.
-The settings in this file override or augment the board configuration settings.
+Setting Application Configuration Values
+========================================
 
-The board configuration settings can be viewed
-in :file:`$ZEPHYR_BASE/boards/ARCHITECTURE/BOARD/BOARD_defconfig`.
+This section describes how to edit Zephyr configuration
+(:file:`.conf`) files.
 
-.. note::
+- Add each configuration entry on a new line.
+
+- Enable a boolean option by setting its value to ``y``:
 
-   When the default board configuration settings are sufficient for your
-   application, a :file:`.conf` file is not needed. Skip ahead to
-   :ref:`override_kernel_conf`.
+  .. code-block:: none
 
+     CONFIG_SOME_BOOL=y
 
-#. Navigate to the :file:`app`, and create the :file:`prj.conf` file. Enter:
+  To ensure that a boolean configuration option is not set, add a line
```

Comment:
are you having them add a line, or are then commenting out an existing line?

Reply:
This is one of the subtleties of Kconfig.

The correct approach is to add a line which looks like a comment, but in fact ensures the option is not set.